### PR TITLE
Add barrier queue family index validation and fix 2270

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Refactor and add image/buffer barrier queue family VUIDs and unit tests. Fixing #2270

Refactor existing image barrier VUIDs for source and destination queue family index validation to include all queue family tests.  Refactor validation enables reuse by buffers, all VUID's have parallel usages.

An existing test was removed with all testing for barrier queue families consolidated into an uber-test-case, using a helper object

    VUIDs refactored(*)/new:
    VALIDATION_ERROR_0a000960*   VUID-VkImageMemoryBarrier-image-01200
    VALIDATION_ERROR_0a000962    VUID-VkImageMemoryBarrier-image-01201
    VALIDATION_ERROR_0a00096a    VUID-VkImageMemoryBarrier-image-01205
    VALIDATION_ERROR_0a000dce    VUID-VkImageMemoryBarrier-image-01767
    VALIDATION_ERROR_0a000dd0    VUID-VkImageMemoryBarrier-image-01768
    VALIDATION_ERROR_0a00095e*   VUID-VkImageMemoryBarrier-image-01199
    VALIDATION_ERROR_0a000aca    VUID-VkImageMemoryBarrier-image-01381
    VALIDATION_ERROR_0a000dcc    VUID-VkImageMemoryBarrier-image-01766

    VALIDATION_ERROR_0180094e    VUID-VkBufferMemoryBarrier-buffer-01191
    VALIDATION_ERROR_01800dc6    VUID-VkBufferMemoryBarrier-buffer-01763
    VALIDATION_ERROR_01800952    VUID-VkBufferMemoryBarrier-buffer-01193
    VALIDATION_ERROR_01800dca    VUID-VkBufferMemoryBarrier-buffer-01765
    VALIDATION_ERROR_01800dc8    VUID-VkBufferMemoryBarrier-buffer-01764
    VALIDATION_ERROR_0180094c    VUID-VkBufferMemoryBarrier-buffer-01190
    VALIDATION_ERROR_01800950    VUID-VkBufferMemoryBarrier-buffer-01192
    VALIDATION_ERROR_01800958    VUID-VkBufferMemoryBarrier-buffer-01196

### CODE REVIEW

**Comment by tobine** [comment]
looks like clang-format check failed

**Comment by jzulauf-lunarg** [comment]
Cleaned formatting... ready to rock.

**Comment by jzulauf-lunarg** [comment]
IGNORE... format fixes incorrectly applied.

**Comment by jzulauf-lunarg** [comment]
Good for review now. Sorry for the noise.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7039,7 +7039,258 @@ static bool ValidateAccessMaskPipelineStage(VkAccessFlags access_mask, VkPipelin
...
+          limit_(static_cast<uint32_t>(device_data->phys_dev_properties.queue_family_properties.size())),
+          mem_ext_(device_data->extensions.vk_khr_external_memory) {}
+
+    // Create a validator state from an image state... reducing the image specific to the generic version.
+    ValidatorState(const layer_data *device_data, const char *func_name, const GLOBAL_CB_NODE *cb_state,
+                   const VkImageMemoryBarrier *barrier, const IMAGE_STATE *state)
+        : ValidatorState(device_data, func_name, cb_state, HandleToUint64(barrier->image), state->createInfo.sharingMode,
+                         kVulkanObjectTypeImage, image_error_codes) {}
+
+    // Create a validator state from an image state... reducing the image specific to the generic version.
```
Comment (daveh-lunarg) [inline layers/core_validation.cpp]: Comment should say 'buffer'

**Pull request merged**